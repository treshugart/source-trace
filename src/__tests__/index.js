const path = require("path");
const { trace } = require("..");

const numFiles = 2;

test("schema", async () => {
  const deps = await trace("!loader?query!./src?query");
  expect(deps).toMatchObject([
    {
      loaders: [],
      originalPath: "./parse-meta",
      path: `./parse-meta`,
      parents: [path.join(process.cwd(), "src", "index.js")],
      query: null,
      resolvedFrom: path.join(process.cwd(), "src"),
      resolvedPath: path.join(process.cwd(), "src", "parse-meta.js")
    },
    {
      loaders: [{ path: "loader", query: "query" }],
      originalPath: "!loader?query!./src?query",
      parents: [],
      path: `./src`,
      query: "query",
      resolvedFrom: process.cwd(),
      resolvedPath: path.join(process.cwd(), "src", "index.js")
    }
  ]);
});

test("ignore default", async () => {
  const deps = await trace("./src");
  expect(deps.length).toBe(numFiles);
});

test("ignore everything", async () => {
  const deps = await trace("./src", { ignore: () => true });
  expect(deps.length).toBe(0);
});

test("project", async () => {
  const deps = await trace("./src/__fixtures__");
  expect(deps.length).toBe(4);
  expect(deps[0].parents.length).toBe(2);
});
