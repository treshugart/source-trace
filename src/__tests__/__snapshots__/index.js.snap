// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults (should ignore deps) 1`] = `
Array [
  "/src/parse-meta.js",
  "/src/index.js",
]
`;

exports[`ignore everything 1`] = `Array []`;

exports[`ignore nothing 1`] = `
Array [
  "/node_modules/find-root/index.js",
  "/node_modules/babylon/lib/index.js",
  "/node_modules/node-source-walk/index.js",
  "/node_modules/ast-module-types/index.js",
  "/node_modules/module-definition/index.js",
  "/node_modules/ms/index.js",
  "/node_modules/debug/src/debug.js",
  "/node_modules/debug/src/browser.js",
  "/node_modules/has-flag/index.js",
  "/node_modules/supports-color/index.js",
  "/node_modules/debug/src/node.js",
  "/node_modules/debug/src/index.js",
  "/node_modules/detective-cjs/index.js",
  "/node_modules/estraverse/estraverse.js",
  "/node_modules/esutils/lib/ast.js",
  "/node_modules/esutils/lib/code.js",
  "/node_modules/esutils/lib/keyword.js",
  "/node_modules/esutils/lib/utils.js",
  "/node_modules/amdefine/amdefine.js",
  "/node_modules/source-map/lib/source-map/base64.js",
  "/node_modules/source-map/lib/source-map/base64-vlq.js",
  "/node_modules/source-map/lib/source-map/util.js",
  "/node_modules/source-map/lib/source-map/array-set.js",
  "/node_modules/source-map/lib/source-map/mapping-list.js",
  "/node_modules/source-map/lib/source-map/source-map-generator.js",
  "/node_modules/source-map/lib/source-map/binary-search.js",
  "/node_modules/source-map/lib/source-map/basic-source-map-consumer.js",
  "/node_modules/source-map/lib/source-map/indexed-source-map-consumer.js",
  "/node_modules/source-map/lib/source-map/source-map-consumer.js",
  "/node_modules/source-map/lib/source-map/source-node.js",
  "/node_modules/source-map/lib/source-map.js",
  "/node_modules/escodegen/package.json",
  "/node_modules/escodegen/escodegen.js",
  "/node_modules/get-amd-module-type/index.js",
  "/node_modules/detective-amd/index.js",
  "/node_modules/detective-es6/index.js",
  "/node_modules/gonzales-pe/lib/gonzales.js",
  "/node_modules/detective-less/node_modules/ms/index.js",
  "/node_modules/detective-less/node_modules/debug/debug.js",
  "/node_modules/detective-less/node_modules/debug/node.js",
  "/node_modules/detective-less/index.js",
  "/node_modules/detective-sass/node_modules/ms/index.js",
  "/node_modules/detective-sass/node_modules/debug/debug.js",
  "/node_modules/detective-sass/node_modules/debug/node.js",
  "/node_modules/detective-sass/index.js",
  "/node_modules/detective-scss/node_modules/ms/index.js",
  "/node_modules/detective-scss/node_modules/debug/debug.js",
  "/node_modules/detective-scss/node_modules/debug/node.js",
  "/node_modules/detective-scss/index.js",
  "/node_modules/detective-stylus/index.js",
  "/node_modules/typescript-eslint-parser/lib/ast-node-types.js",
  "/node_modules/typescript/lib/typescript.js",
  "/node_modules/object-assign/index.js",
  "/node_modules/lodash.tostring/index.js",
  "/node_modules/lodash.unescape/index.js",
  "/node_modules/typescript-eslint-parser/lib/ast-converter.js",
  "/node_modules/typescript-eslint-parser/package.json",
  "/node_modules/typescript-eslint-parser/parser.js",
  "/node_modules/core-js/library/modules/_add-to-unscopables.js",
  "/node_modules/core-js/library/modules/_iter-step.js",
  "/node_modules/core-js/library/modules/_iterators.js",
  "/node_modules/core-js/library/modules/_cof.js",
  "/node_modules/core-js/library/modules/_iobject.js",
  "/node_modules/core-js/library/modules/_defined.js",
  "/node_modules/core-js/library/modules/_to-iobject.js",
  "/node_modules/core-js/library/modules/_library.js",
  "/node_modules/core-js/library/modules/_global.js",
  "/node_modules/core-js/library/modules/_core.js",
  "/node_modules/core-js/library/modules/_a-function.js",
  "/node_modules/core-js/library/modules/_ctx.js",
  "/node_modules/core-js/library/modules/_is-object.js",
  "/node_modules/core-js/library/modules/_an-object.js",
  "/node_modules/core-js/library/modules/_fails.js",
  "/node_modules/core-js/library/modules/_descriptors.js",
  "/node_modules/core-js/library/modules/_dom-create.js",
  "/node_modules/core-js/library/modules/_ie8-dom-define.js",
  "/node_modules/core-js/library/modules/_to-primitive.js",
  "/node_modules/core-js/library/modules/_object-dp.js",
  "/node_modules/core-js/library/modules/_property-desc.js",
  "/node_modules/core-js/library/modules/_hide.js",
  "/node_modules/core-js/library/modules/_export.js",
  "/node_modules/core-js/library/modules/_redefine.js",
  "/node_modules/core-js/library/modules/_has.js",
  "/node_modules/core-js/library/modules/_to-integer.js",
  "/node_modules/core-js/library/modules/_to-length.js",
  "/node_modules/core-js/library/modules/_to-absolute-index.js",
  "/node_modules/core-js/library/modules/_array-includes.js",
  "/node_modules/core-js/library/modules/_shared.js",
  "/node_modules/core-js/library/modules/_uid.js",
  "/node_modules/core-js/library/modules/_shared-key.js",
  "/node_modules/core-js/library/modules/_object-keys-internal.js",
  "/node_modules/core-js/library/modules/_enum-bug-keys.js",
  "/node_modules/core-js/library/modules/_object-keys.js",
  "/node_modules/core-js/library/modules/_object-dps.js",
  "/node_modules/core-js/library/modules/_html.js",
  "/node_modules/core-js/library/modules/_object-create.js",
  "/node_modules/core-js/library/modules/_wks.js",
  "/node_modules/core-js/library/modules/_set-to-string-tag.js",
  "/node_modules/core-js/library/modules/_iter-create.js",
  "/node_modules/core-js/library/modules/_to-object.js",
  "/node_modules/core-js/library/modules/_object-gpo.js",
  "/node_modules/core-js/library/modules/_iter-define.js",
  "/node_modules/core-js/library/modules/es6.array.iterator.js",
  "/node_modules/core-js/library/modules/web.dom.iterable.js",
  "/node_modules/core-js/library/modules/_string-at.js",
  "/node_modules/core-js/library/modules/es6.string.iterator.js",
  "/node_modules/core-js/library/modules/_classof.js",
  "/node_modules/core-js/library/modules/core.get-iterator-method.js",
  "/node_modules/core-js/library/modules/core.get-iterator.js",
  "/node_modules/core-js/library/fn/get-iterator.js",
  "/node_modules/babel-runtime/core-js/get-iterator.js",
  "/node_modules/core-js/library/modules/_object-sap.js",
  "/node_modules/core-js/library/modules/es6.object.get-prototype-of.js",
  "/node_modules/core-js/library/fn/object/get-prototype-of.js",
  "/node_modules/babel-runtime/core-js/object/get-prototype-of.js",
  "/node_modules/babel-runtime/helpers/classCallCheck.js",
  "/node_modules/core-js/library/modules/es6.object.define-property.js",
  "/node_modules/core-js/library/fn/object/define-property.js",
  "/node_modules/babel-runtime/core-js/object/define-property.js",
  "/node_modules/babel-runtime/helpers/createClass.js",
  "/node_modules/core-js/library/modules/_wks-ext.js",
  "/node_modules/core-js/library/fn/symbol/iterator.js",
  "/node_modules/babel-runtime/core-js/symbol/iterator.js",
  "/node_modules/core-js/library/modules/_meta.js",
  "/node_modules/core-js/library/modules/_wks-define.js",
  "/node_modules/core-js/library/modules/_object-gops.js",
  "/node_modules/core-js/library/modules/_object-pie.js",
  "/node_modules/core-js/library/modules/_enum-keys.js",
  "/node_modules/core-js/library/modules/_is-array.js",
  "/node_modules/core-js/library/modules/_object-gopn.js",
  "/node_modules/core-js/library/modules/_object-gopn-ext.js",
  "/node_modules/core-js/library/modules/_object-gopd.js",
  "/node_modules/core-js/library/modules/es6.symbol.js",
  "/node_modules/core-js/library/modules/es6.object.to-string.js",
  "/node_modules/core-js/library/modules/es7.symbol.async-iterator.js",
  "/node_modules/core-js/library/modules/es7.symbol.observable.js",
  "/node_modules/core-js/library/fn/symbol/index.js",
  "/node_modules/babel-runtime/core-js/symbol.js",
  "/node_modules/babel-runtime/helpers/typeof.js",
  "/node_modules/babel-runtime/helpers/possibleConstructorReturn.js",
  "/node_modules/core-js/library/modules/_set-proto.js",
  "/node_modules/core-js/library/modules/es6.object.set-prototype-of.js",
  "/node_modules/core-js/library/fn/object/set-prototype-of.js",
  "/node_modules/babel-runtime/core-js/object/set-prototype-of.js",
  "/node_modules/core-js/library/modules/es6.object.create.js",
  "/node_modules/core-js/library/fn/object/create.js",
  "/node_modules/babel-runtime/core-js/object/create.js",
  "/node_modules/babel-runtime/helpers/inherits.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/util/identifier.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/options.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/tokenizer/types.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/util/whitespace.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/tokenizer/context.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/util/location.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/tokenizer/state.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/tokenizer/index.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/index.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/util.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/statement.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/lval.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/expression.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/node.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/location.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/parser/comments.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/plugins/flow.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/plugins/jsx/xhtml.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/plugins/jsx/index.js",
  "/node_modules/detective-typescript/node_modules/babylon/lib/index.js",
  "/node_modules/detective-typescript/node_modules/node-source-walk/index.js",
  "/node_modules/detective-typescript/index.js",
  "/node_modules/precinct/index.js",
  "/node_modules/resolve/lib/core.json",
  "/node_modules/resolve/lib/core.js",
  "/node_modules/resolve/lib/caller.js",
  "/node_modules/path-parse/index.js",
  "/node_modules/resolve/lib/node-modules-paths.js",
  "/node_modules/resolve/lib/async.js",
  "/node_modules/resolve/lib/sync.js",
  "/node_modules/resolve/index.js",
  "/src/parse-meta.js",
  "/src/index.js",
]
`;
